acsf_tools.drush.inc

```diff
@@ -179,46 +179,56 @@ function drush_acsf_tools_ml() {
-
-        $site_settings_filepath = 'sites/g/files/' . $details['name'] . '/settings.php';
-        if (file_exists($site_settings_filepath)) {
-          $site_settings = @file_get_contents($site_settings_filepath);
-          if (preg_match("/'install_profile'] = '([a-zA-Z_]*)'/", $site_settings, $matches)) {
-            if (isset($matches[1]) && !in_array($matches[1], $profiles)) {
+    $total_time_limit = (int) drush_get_option('total-time-limit');
+
+    $end = time() + $total_time_limit;
+    do {
+      foreach ($sites as $delta => $details) {
+        // Get the first custom domain if any. Otherwise use the first domain
+        // which is *.acsitefactory.com. Given this is used as --uri parameter
+        // by the drush command, it can have an impact on the drupal process.
+        $domain = $details['domains'][1] ?? $details['domains'][0];
+
+        // If specific profile is provided, we check if it is the site's profile.
+        // We first try to get site's profile from the settings file. If we can't
+        // find the settings file, we use drush core-status to get it.
+        if (!empty($profiles)) {
+          $proceed = TRUE;
+
+          $site_settings_filepath = 'sites/g/files/' . $details['name'] . '/settings.php';
+          if (file_exists($site_settings_filepath)) {
+            $site_settings = @file_get_contents($site_settings_filepath);
+            if (preg_match("/'install_profile'] = '([a-zA-Z_]*)'/", $site_settings, $matches)) {
+              if (isset($matches[1]) && !in_array($matches[1], $profiles)) {
+                $proceed = FALSE;
+              }
+            }
+          }
+          else {
+            $profile_options = $options;
+            $profile_options['fields'] = 'install-profile';
+            $profile_options['l'] = $domain;
+            $site_profile_info = drush_invoke_process('@self', 'core-status', [], $profile_options, ['output' => FALSE]);
+            if (!in_array($site_profile_info['object']['install-profile'], $profiles)) {
               $proceed = FALSE;
             }
           }
-        }
-        else {
-          $profile_options = $options;
-          $profile_options['fields'] = 'install-profile';
-          $profile_options['l'] = $domain;
-          $site_profile_info = drush_invoke_process('@self', 'core-status', [], $profile_options, ['output' => FALSE]);
-          if (!in_array($site_profile_info['object']['install-profile'], $profiles)) {
-            $proceed = FALSE;
+
+          if (!$proceed) {
+            drush_log("\n=> Skipping command on $domain", 'ok');
+            continue;
           }
         }
 
-        if (!$proceed) {
-          drush_log("\n=> Skipping command on $domain", 'ok');
-          continue;
+        drush_log("\n=> Running command on $domain", 'ok');
+        $output = drush_invoke_process('@self', $command, $arguments, $options + array('l' => $domain));
+
+        // @TODO Find a better way to remove sites from execution list.
```

Comment:
This is likely not generic enough given that we could run any drush command.
I would rather add an optional parameter called time-limit-site-removal-condition that would contain a regexp. If drush output matches that regexp then the site is removed from the list of sites.

Reply:
@jose-oliveira how essential is this in the implementation? This looks to me like a nice to have improvement for your very specific use case. So I am tempted to remove from this PR.